core/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -79,6 +79,16 @@ public final Query toQuery(QueryParseContext parseContext) throws IOException {
         return query;
     }
 
+    /**
+     * Few queries allow you to set the boost and queryName but don't do anything with it. This method allows
+     * to disable boost and queryName for those queries.
+     */
+    //norelease this has to be put in here because of the call on toQuery which would override the boost and queryName
+    //norelease of the wrapped query with defaults
+    protected boolean supportsBoostAndQueryName() {
```

Comment:
AbstractQueryBuilder#toQuery does set a boost unless it is disabled. If I remember correctly I think that was the idea of being to disable boost as part of the actual query builder, not just in the test.